zebra/zebra_nhg.h

```diff
@@ -102,20 +102,23 @@ struct nhg_hash_entry {
  * Is this a nexthop that is recursively resolved?
  */
 #define NEXTHOP_GROUP_RECURSIVE (1 << 3)
-/*
- * This is a nexthop group we got from the kernel, it is identical to
- * one we already have. (The kernel allows duplicate nexthops, we don't
- * since we hash on them). We are only tracking it in our ID table,
- * it is unusable by our created routes but may be used by routes we get
- * from the kernel. Therefore, it is unhashable.
- */
-#define NEXTHOP_GROUP_UNHASHABLE (1 << 4)
```

Comment:
how does support for user-created nhgs (using iproute e.g.) work now? for example, IGP sends a route, nhg gets created by hashing, and then someone changes the group (by id) with iproute - are we able to handle that, to notice that the group belongs to zebra and needs to match what was sent with a route?

Reply:
we only track external NHGs in the table to match with routes using them. We dont actually do anything with them internally.

If we want to use a NHG thats exactly the same, it will get a new ID.